hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/model/CellModel.java

```diff
@@ -218,9 +218,10 @@ public Message messageFromObject() {
   }
 
   @Override
-  public ProtobufMessageHandler getObjectFromMessage(byte[] message) throws IOException {
+  public ProtobufMessageHandler getObjectFromMessage(CodedInputStream cis) throws IOException {
     Cell.Builder builder = Cell.newBuilder();
-    ProtobufUtil.mergeFrom(builder, message);
+    builder.mergeFrom(cis);
```

Comment:
Better still introduce a method in ProtobufUtil to put these two lines? Otherwise we need to repeat these two lines in every classes. And why we need to call checkLastTagWas?

Reply:
We can't use ProtobufUtil because of shading conflicts on branch-2.x.
But I can add method to RestUtil.

As for why call checkLastTagWas: Frankly, that's what the orginal code did.
I think that the intention was  to have extra check to make double sure that we have read the data fully.
It does not look epensive, so I left it in, but we can just remove it, and trust the error handling of the http libraries to throw an exception in case we don't get the full data.

Reply:
It is OK to just keep it as is, just asking...